Sofa/framework/Geometry/src/sofa/geometry/Edge.h

```diff
@@ -159,6 +204,109 @@ struct Edge
+    * @tparam   Node iterable container
+    * @tparam   T scalar
+    * @param	pA, pB nodes of the first edge
+    * @param	pC, pD nodes of the second edge
+    * @param    intersection node will be filled if there is an intersection otherwise will return sofa::InvalidID
+    * @return	bool true if there is an intersection, otherwise false
+    */
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>
+    >
+        [[nodiscard]]
+    static constexpr bool intersectionWithEdge(const Node& pA, const Node& pB, const Node& pC, const Node& pD, Node& intersection)
+    {
+        // The 2 segment equations using pX on edge1 and pY on edge2 can be defined by:
+        // pX = pA + alpha (pB - pA)
+        // pY = pC + beta (pD - pC)
+        const auto AB = pB - pA;
+        const auto CD = pD - pC;
+
+        if constexpr (std::is_same_v < Node, sofa::type::Vec<2, T> >)
+        {
+            // in 2D we have 2 segment equations and 2 unknowns so direct solving of pX = pY is possible
+            // pA + alpha (pB - pA) = pC + beta (pD - pC)
+            // alpha = ((Cy - Ay)(Dx - Cx) - (Cx - Ax)(Dy - Cy)) / ((By - Ay)(Dx - Cx) - (Bx - Ax)(Dy - Cy))
+            const auto AC = pC - pA;
+            const T alphaNom = AC[1] * CD[0] - AC[0] * CD[1];
+            const T alphaDenom = AB[1] * CD[0] - AB[0] * CD[1];
+            
+            if (alphaDenom < std::numeric_limits<T>::epsilon()) // collinear
+            {
+                intersection = sofa::type::Vec<2, T>(sofa::InvalidID, sofa::InvalidID);
+                return false;
+            }
+            
+            const T alpha = alphaNom / alphaDenom;
+
+            if (alpha < 0 || alpha > 1)
+            {
+                intersection = sofa::type::Vec<2, T>(sofa::InvalidID, sofa::InvalidID);
+                return false;
+            }
+            else
+            {
+                intersection = pA + alpha * AB;
+                return true;
+            }
+        }
+        else
+        {
+            // We search for the shortest line between the two 3D lines. If this lines length is null then there is an intersection
+            // Shortest segment [pX; pY] between the two lines will be perpendicular to them. Then:
+            // (pX - pY).dot(pB - pA) = 0
+            // (pX - pY).dot(pD - pC) = 0
+            
+            // We need to find alpha and beta that suits: 
+            // [ (pA - pC) + alpha(pB - pA) - beta(pD - pC) ].dot(pB - pA) = 0
+            // [ (pA - pC) + alpha(pB - pA) - beta(pD - pC) ].dot(pD - pC) = 0
+            const auto CA = pA - pC;
+
+            // Writting d[CA/AB] == (pA - pC).dot(pB - pA) and subtituting beta we obtain:
+            // beta = (d[CA/CD] + alpha * d[AB/CD]) / d[CD/CD]
+            // alpha = ( d[CA/CD]*d[CD/AB] - d[CA/AB]*d[CD/CD] ) / ( d[AB/AB]*d[CD/CD] - d[AB/CD]*d[AB/CD])
```

Comment:
I tried to do the math on my side and it's good !

Reply:
ouf! 😄